kernels/volk/volk_8i_x2_saturated_sum_8i.h

```diff
@@ -0,0 +1,277 @@
+  const unsigned int vecNum = num_vals / VEC_SIZE;
+  unsigned int i = 0;
+  int16_t tmpPlus; // Need to move to more bits to workaround possible overflow
+
+  for(; i < vecNum; i++) {
+    __m128i a = _mm_load_si128((const __m128i*)(inVectorA + VEC_SIZE*i));
+    __m128i b = _mm_load_si128((const __m128i*)(inVectorB + VEC_SIZE*i));
+    __m128i result = _mm_adds_epi8(b, a);
+    _mm_store_si128((__m128i*)(outVector + VEC_SIZE*i), result);
+  }
+
+  i = VEC_SIZE*vecNum;
+  for(; i < num_vals; i++) {
+    tmpPlus = inVectorB[i] + inVectorA[i];
+    // Saturate to avoid overflow
+    tmpPlus = (tmpPlus > 127) ? 127 : tmpPlus;
+    tmpPlus = (tmpPlus < -128) ? -128 : tmpPlus;
+
+    outVector[i] = ((int8_t)tmpPlus);
+  }
+}
+
+#endif /* LV_HAVE_SSE2 for aligned */
+
+
+#ifdef LV_HAVE_AVX2
+
+#include <x86intrin.h>
+
+static inline void
+volk_8i_x2_saturated_sum_8i_a_avx2(int8_t* outVector,
+                                   const int8_t* inVectorA, const int8_t* inVectorB,
+                                   const unsigned int num_vals)
+{
+  const unsigned int VEC_SIZE = 32;
+  const unsigned int vecNum = num_vals / VEC_SIZE;
+  unsigned int i = 0;
+  int16_t tmpPlus; // Need to move to more bits to workaround possible overflow
+
+  for(; i < vecNum; i++) {
+    __m256i a = _mm256_load_si256((const __m256i*)(inVectorA + VEC_SIZE*i));
+    __m256i b = _mm256_load_si256((const __m256i*)(inVectorB + VEC_SIZE*i));
+    __m256i result = _mm256_adds_epi8(b, a);
+    _mm256_store_si256((__m256i*)(outVector + VEC_SIZE*i), result);
+  }
+
+  i = VEC_SIZE*vecNum;
+  for(; i < num_vals; i++) {
+    tmpPlus = inVectorB[i] + inVectorA[i];
+    // Saturate to avoid overflow
+    tmpPlus = (tmpPlus > 127) ? 127 : tmpPlus;
+    tmpPlus = (tmpPlus < -128) ? -128 : tmpPlus;
+
+    outVector[i] = ((int8_t)tmpPlus);
+  }
+}
+
+#endif /* LV_HAVE_AVX2 for aligned */
+
+
+#ifdef LV_HAVE_NEON
+
+#include <arm_neon.h>
+
+static inline void
+volk_8i_x2_saturated_sum_8i_a_neon(int8_t* outVector,
+                                   const int8_t* inVectorA, const int8_t* inVectorB,
+                                   const unsigned int num_vals)
+{
+  const unsigned int VEC_SIZE = 16;
+  const unsigned int vecNum = num_vals / VEC_SIZE;
+  unsigned int i = 0;
+  int16_t tmpPlus; // Need to move to more bits to workaround possible overflow
+
+  for(; i < vecNum; i++) {
+		int8x16_t a = vld1q_s8((const int8x16_t*)(inVectorA + VEC_SIZE*i));
+		int8x16_t b = vld1q_s8((const int8x16_t*)(inVectorB + VEC_SIZE*i));
+		vst1q_s8((const int8x16_t*)(outVector + VEC_SIZE*i), vqaddq_s8(b, a));
+  }
+
+  i = VEC_SIZE*vecNum;
+  for(; i < num_vals; i++) {
+    tmpPlus = inVectorB[i] + inVectorA[i];
+    // Saturate to avoid overflow
+    tmpPlus = (tmpPlus > 127) ? 127 : tmpPlus;
+    tmpPlus = (tmpPlus < -128) ? -128 : tmpPlus;
+
+    outVector[i] = ((int8_t)tmpPlus);
+  }
+}
+
+#endif /* LV_HAVE_NEON for aligned */
+
+#endif /* INCLUDED_VOLK_8s_SATURATED_SUM_8s_ALIGNED8_H */
```

Comment:
this #endif should be `INCLUDED_volk_8i_x2_saturated_sum_8i_u_H` ... same as that starting the header. case matters!